src/components/navigation/GlobalSettingsButton.tsx

```diff
@@ -0,0 +1,98 @@
+import LanguageIcon from '@mui/icons-material/Language';
+import { IconButton, List, ListItem, ListItemButton, Popover } from '@mui/material';
+import { useCallback, useState } from 'react';
+import toast from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+
+import useEditAccountMutation from '@src/api/mutations/account/useEditAccountMutation';
+import { LOCAL_STORAGE_KEYS } from '@src/config/local-storage.config';
+import { LANGUAGE_LIST } from '@src/constants';
+import { LanguageCollection } from '@src/constants/constant.types';
+import { useAppSelector } from '@src/hooks/redux-hooks';
+import i18n from '@src/i18n/i18n';
+import { ErrorTypes } from '@src/types/i18n.types';
+import { setValueToLocalStorage } from '@src/utils';
+
+const GlobalSettingsButton = () => {
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const open = Boolean(anchorEl);
+  const { t } = useTranslation();
+
+  const profile = useAppSelector((state) => state.auth?.user?.profile);
+  const profileId = profile?.id ?? '';
+  const { mutate } = useEditAccountMutation(profileId);
+
+  const handleOpen = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleLangChange = useCallback(
+    async (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+      const dataset = event.currentTarget.dataset;
+      const code = dataset.code as LanguageCollection['code'];
+
+      try {
+        if (profile && code !== profile?.language) {
+          mutate({ language: code });
+        }
+
+        await i18n.changeLanguage(code);
```

Comment:
do you need these 3 operations outside if block? Should they be included in if block?

Reply:
basically, in if block I'm checking that user has a profile, if yes I'm sending request to change laguage property of the profile.
below statements should be executed regardles of the users profile (in case, user hasn't logged in, changes will be only locally)